### PR TITLE
Add additional log checkpoint information

### PR DESCRIPTION
**Pull request by directionless**
In thinking about how to add more information to the checkpoint logs it felt like this needed to become a struct -- easier to hold some state, add a querier, and whatnot.

1. Refactor to a checkpointer struct, moved the `Run` around a little
2. Add a `SetQuerier` function
3. When there is a querier, additional things are queried. 

Fixes: #1061

### CODE REVIEW

File: pkg/log/checkpoint/queried.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+const osquerySqlQuery = `
+SELECT
+	osquery_info.version as osquery_version,
+	osquery_info.instance_id as osquery_instance_id
+FROM
+    osquery_info
+`
+
+func (c *checkPointer) logOsqueryInfo() {
```
Comment (James-Pickett) [inline pkg/log/checkpoint/queried.go:38]: should we add some data from `kolide_launcher_osquery_instance_history`? could possibly help up identify if osquery is crashing often and why

**Review by James-Pickett — APPROVED** [review]
Nice!

File: pkg/log/checkpoint/queried.go (hunk 1 above)
Comment (directionless) [reply pkg/log/checkpoint/queried.go:38]: That's a good thought. I'm going to merge that, and flag that for the next pass

Comment (directionless) [reply pkg/log/checkpoint/queried.go:38]: https://github.com/kolide/launcher/issues/1071

**Pull request merged**